php_phongo.c

```diff
@@ -1638,34 +1627,32 @@ static void phongo_result_iterator_move_forward(zend_object_iterator *iter TSRML
 
 			bson_iter_document(&cursor_it->first_batch_iter, &data_len, &data);
 
-			MAKE_STD_ZVAL(result->visitor_data.zchild);
-			bson_to_zval(data, data_len, &result->visitor_data);
+			MAKE_STD_ZVAL(cursor->visitor_data.zchild);
+			bson_to_zval(data, data_len, &cursor->visitor_data);
 			return;
 		}
 	}
-	if (mongoc_cursor_next(result->cursor, &doc)) {
-		MAKE_STD_ZVAL(result->visitor_data.zchild);
-		bson_to_zval(bson_get_data(doc), doc->len, &result->visitor_data);
+	if (mongoc_cursor_next(cursor->cursor, &doc)) {
+		MAKE_STD_ZVAL(cursor->visitor_data.zchild);
+		bson_to_zval(bson_get_data(doc), doc->len, &cursor->visitor_data);
 	} else {
-		iter->funcs->invalidate_current(iter TSRMLS_CC);
+		/* TODO: is this really necessary? */
```

Comment:
The reason I didn't think it was necessary was that we already call `php_phongo_cursor_free_current()` at the top of this function. Thereafter, the `cursor->visitor_data` is only populated in two possible places:

 * `bson_iter_next()` on `first_batch_iter` succeeded, and the element was a document. `bson_to_zval()` is invoked to populate `cursor->visitor_data` and we return immediately.
 * `mongoc_cursor_next()` returned successfully and sets the `doc` pointer. We then call `bson_to_zval()` to populate `cursor->visitor_data`.

In the latter case (cursor iteration), this line of code is the `else` condition (i.e. cursor iteration didn't have a next document). I don't see any point where `cursor->visitor_data` could have been populated.